src/snowflake/snowpark/files.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+"""SnowflakeFile for UDFs in Snowpark."""
+from __future__ import annotations
+
+import array
+from typing import Iterable
+
+_DEFER_IMPLEMENTATION_ERR_MSG = (
+    "The file operations is implemented in Snowflake servers"
+)
+
+
+class SnowflakeFile:
+    """
+    SnowflakeFile provides developers an interface to operate files as a Python IOBase-like object in UDFs.
+    SnowflakeFile supports most operations supported by Python IOBase
+    For more information about the class, please check ...(docs TBD)
+
+    The constructor of this class is not supposed to be called directly. Call :meth:`~snowflake.snowpark.file.SnowflakeFile.open` to create an instance of SnowflakeFile. It can be used as a python file object.
+    The input type can be a column name as a :class:`str`, or a :class:`~snowflake.snowpark.Column` object.
+    """
+
+    def __init__(
+        self,
+        file_location: str,
+        mode: str = "r",
+    ) -> None:
+        # The URL/URI of the file to be opened and wrapped by the SnowflakeFile instance.
+        self._file_location: str = file_location
+        # The mode of file stream
+        self._mode: str = mode
+
+        # The attributes supported as part of IOBase
+        self.buffer = None
+        self.closed = None
+        self.encoding = None
+        self.errors = None
+
+    @classmethod
+    def open(
```

Comment:
Or we can add the decorator here since users always need to call `open` first.

Reply:
I think it is better since if we add a decorator to a class, it will be a class-decorator, which may be hard for other devs to re-use it